tests/integration_test/iam_imds_data_path_test.go

```diff
@@ -113,17 +113,19 @@ func TestIamImdsDataPath(t *testing.T) {
 			}
 
 			if tc.wantErr != "" {
-				// Skip health checks since we expect an error.
 				s.SkipHealthChecks()
-				glog.Infof("Sleeping to ensure Envoy is starting")
-				time.Sleep(7 * time.Second)
 			}
 
 			defer s.TearDown(t)
 			if err := s.Setup(tc.confArgs); err != nil {
 				t.Fatalf("fail to setup test env, %v", err)
 			}
 
+			if tc.wantErr != "" {
```

Comment:
Don't understand why we only need to sleep for wantErr case. Could you add comment on that?

Reply:
Done.